### DISCUSSION TITLE
External Editor recommendations? Single file, single window, no tabs – to work as much an internal one

### DISCUSSION THREAD

**Discussion by WWWFWWW** [mikekazakov/nimble-commander#94] [General]
Any recommendations for an external editor that would work well with NC?

Syntax highlighting and fast speed is a must, obviously.

I am using BBEdit and CotEditor usually, but these are more suited to be standalone editors – open a lot of tabs, edit a lot of files, do a whole project with them.

I'd like a simple "single file – single window – no tabs, no multiple files – save & quit, no reopen" editor that would integrate well with Nimble Commander for quickly editing files one after the other with F4.

I could always use the built‑in Terminal window to run vim, nano or micro to edit the file, setting a Tool command for that, but that doesn't help either, as after saving and quitting the editor, I am still at the terminal prompt instead of the file panel and have to press Hide Terminal shortcut again.

Thanks!

**Comment by WWWFWWW** [comment]
BTW, I can obviously configure CotEditor to never remember open files and not to use tabs, which does work the way I wrote above, but I'd rather keep it the way it is for other use outside NC...

**Comment by mikekazakov** [comment]
But running vim as an editor (not as a tool) does bring the UI back to file panels, right?
That's what I'm personally using when occasionally need to edit something via ssh.
mcedit can be used as well.

**Reply by WWWFWWW** [reply]
Ah, yes – configuring it as an _editor_, not _tool_ works perfectly for terminal editors, thanks! Shows and hides the terminal panel upon quitting the terminal editor nicely.

I am currently using _micro_ for that.

And thanks to Nimble Commander's customisability, I can easily set up another GUI editor using the _tools_, whenever I want full GUI text editor, and configure a separate hotkey like shift-F4 for it (settled on CotEditor with tabs disabled, so far).

I'll leave the topic open, since others might still want to suggest their favourite syntax highlighting text editor best suited for use with NC, either a GUI one or terminal – no emacs/vim wars please, though! ;-)

**Comment by okyr-ux** [comment]
I use TextMate and Tincta for that purpose.
Tincta is a bit outdated, but works just fine. 
TextMate is more like CotEditor, but again, you can configure it as you described above. Otherwise it's an awesome editor.
